api/internal/git/repospec.go

```diff
@@ -219,16 +234,7 @@ func parseQuery(query string) (string, time.Duration, bool) {
 	return ref, duration, submodules
 }
 
-func parsePath(n string) string {
-	parsed, err := url.Parse(n)
-	// TODO(annasong): decide how to handle error, i.e. return error, empty string, etc.
```

Comment:
This PR is suggesting removing this parsing entirely. I noticed that no tests fail if we do so, and so I was wondering what this is intended to catch. I looked back in the history, and it was introduced by https://github.com/kubernetes-sigs/kustomize/pull/3900. IIUC that PR intended it as a more reliable way (vs the former regex) to separate query parts from the path. We still use it for that today; since https://github.com/kubernetes-sigs/kustomize/pull/4863 we just do it a bit earlier in the parsing flow. At that earlier point, we're passing the whole raw value to the parser, whereas originally (in valid cases anyway) we were passing just the kust root path + query. So the Path value we would get back at our new invocation is not usable. 

The question is, do we then need to continue to put the path through the URL parser at the end of the parsing flow, now on its own (the query long since removed)? I don't think so, because IIUC the original intent was not to validate the path; it was just convenient to take the value returned when we separated a query from it. If you go back one more step in the history, peelQuery was just splitting a query chunk off of `n` when present.

This also relates to the stance I explained on my last PR re: passing things through to git if they're not interfering with our ability to parse out the repo root. In this case, we've necessarily already done that.

LMK if you agree.